src/dask_awkward/lib/core.py

```diff
@@ -1530,22 +1564,43 @@ def map_partitions(
     This is effectively the same as `d = c * a`
 
     """
-    token = token or tokenize(fn, *args, meta, **kwargs)
-    label = label or funcname(fn)
+    token = token or tokenize(base_fn, *args, meta, **kwargs)
+    label = label or funcname(base_fn)
```

Comment:
purely cosmetic nit:
```suggestion
    label = hyphenize(label or funcname(base_fn))
```